### PR TITLE
Poll mode support for Server 2025

### PR DESCRIPTION
**Pull request by ybendito**

### CODE REVIEW

File: NetKVM/NetKVM-VS2015.vcxproj [hunk 1]
```diff
@@ -170,7 +170,7 @@
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)'=='Win11 Release'">
     <ClCompile>
-      <PreprocessorDefinitions>%(PreprocessorDefinitions);NDIS685_MINIPORT=1;NTDDI_VERSION=0xA0000004</PreprocessorDefinitions>
+      <PreprocessorDefinitions>%(PreprocessorDefinitions);NDIS686_MINIPORT=1;NTDDI_VERSION=0xA0000004</PreprocessorDefinitions>
```
[the remark this answers is not in the document]
Comment (YanVugenfirer) [reply NetKVM/NetKVM-VS2015.vcxproj:173]: 1. We will need to update this on the actual eWDK.

File: NetKVM/Common/ParaNdis_Common.cpp [hunk 2]
```diff
@@ -1320,6 +1324,26 @@ NDIS_STATUS ParaNdis_FinishInitialization(PARANDIS_ADAPTER *pContext)
...
+    if (status == NDIS_STATUS_SUCCESS && pContext->bPollModeTry &&
+        pContext->RSSMaxQueuesNumber && pContext->bRSSOffloadSupported &&
+        (UINT)pContext->RSSMaxQueuesNumber >= pContext->nPathBundles)
+    {
+        int nPollModeOK = 0;
+        for (int i = 0; i < pContext->RSSMaxQueuesNumber; i++)
+        {
+            nPollModeOK += pContext->PollHandlers[i].Register(pContext, i);
+        }
+        if (nPollModeOK == pContext->RSSMaxQueuesNumber)
```
[the remark this answers is not in the document]
Comment (YanVugenfirer) [reply NetKVM/Common/ParaNdis_Common.cpp:1336]: The amount of RSS queues and bundles should be investigated later

File: NetKVM/Common/Trace.h [hunk 3]
```diff
@@ -19,6 +19,8 @@
 #define TraceNoPrefix DPrintf
 #endif
 
+#define POLL_PRINT_LEVEL 2
```
[the remark this answers is not in the document]
Comment (YanVugenfirer) [reply NetKVM/Common/Trace.h:22]: Let's use generic mechanism and move to level 3 in next PR

**Comment by ybendito** [comment]
2 Failed tests on Win2022 (E2E, USO) - not reproduced in my local environment, I suspect there is some problem with HLK env

**Review by YanVugenfirer — APPROVED** [review]

**Pull request merged**